crates/datasources/src/native/access.rs

```diff
@@ -78,36 +79,40 @@ impl NativeTableStorage {
         Ok(total_size)
     }
 
-    pub async fn create_table(&self, table: &TableEntry) -> Result<NativeTable> {
+    pub async fn create_table(&self, table: &TableEntry, or_replace: bool) -> Result<NativeTable> {
         let delta_store = self.create_delta_store_for_table(table).await?;
-
         let opts = Self::opts_from_ent(table)?;
-        let mut builder = CreateBuilder::new()
-            .with_table_name(&table.meta.name)
-            .with_object_store(delta_store)
-            .with_save_mode(SaveMode::ErrorIfExists);
-        for col in &opts.columns {
-            let column = match col.arrow_type.clone() {
-                DataType::Timestamp(_, tz) => InternalColumnDefinition {
-                    name: col.name.clone(),
-                    nullable: col.nullable,
-                    arrow_type: DataType::Timestamp(TimeUnit::Microsecond, tz),
-                },
-                _ => col.to_owned(),
-            };
-            builder = builder.with_column(
-                column.name.clone(),
-                (&column.arrow_type).try_into()?,
-                column.nullable,
-                None,
-            );
-        }
+        let tbl = {
+            let mut builder = CreateBuilder::new()
+                .with_table_name(&table.meta.name)
+                .with_object_store(delta_store);
 
-        // TODO: Partitioning
+            if or_replace {
+                builder = builder.with_save_mode(SaveMode::Overwrite);
+            }
```

Comment:
Seems like we'll want an else here to set save mode to `ErrorIfExists`

Reply:
the default for `CreateBuilder` is `SaveMode::ErrorIfExists` so no need to specify it.